geowebcache/mbtiles/src/test/java/org/geowebcache/mbtiles/layer/MBTilesLayerTest.java

```diff
@@ -0,0 +1,138 @@
+import org.junit.rules.TemporaryFolder;
+
+public class MBTilesLayerTest {
+
+    public @Rule MockWepAppContextRule extensions = new MockWepAppContextRule();
+    protected TileLayerConfiguration config;
+
+    @Before
+    public void setUpTestUnit() throws Exception {
+        config = getConfig();
+    }
+
+    @Rule public TemporaryFolder temp = new TemporaryFolder();
+    protected File configDir;
+    protected File configFile;
+
+    protected TileLayerConfiguration getConfig() throws Exception {
+        makeConfigFile();
+        return getConfig(extensions);
+    }
+
+    private void makeConfigFile() throws Exception {
+        if (configFile == null) {
+            configDir = temp.getRoot();
+            configFile = temp.newFile("geowebcache.xml");
+
+            File lakesFile = temp.newFile("world_lakes.mbtiles");
+            URL source = MBTilesLayerTest.class.getResource("world_lakes.mbtiles");
+            Files.copy(
+                    Paths.get(source.toURI()),
+                    lakesFile.toPath(),
+                    StandardCopyOption.REPLACE_EXISTING);
+
+            File madagascarFile = temp.newFile("madagascar.mbtiles");
+            source = MBTilesLayerTest.class.getResource("madagascar.mbtiles");
+            Files.copy(
+                    Paths.get(source.toURI()),
+                    madagascarFile.toPath(),
+                    StandardCopyOption.REPLACE_EXISTING);
+
+            source = MBTilesLayerTest.class.getResource("geowebcache.xml");
+            try (Stream<String> lines = Files.lines(Paths.get(source.toURI()))) {
+                List<String> replaced =
+                        lines.map(
+                                        line -> {
+                                            return line.replace(
+                                                            "world_lakes.mbtiles",
+                                                            lakesFile.getAbsolutePath())
+                                                    .replace(
+                                                            "madagascar.mbtiles",
+                                                            madagascarFile.getAbsolutePath())
+                                                    .replace("\\", "\\\\");
+                                        })
+                                .collect(Collectors.toList());
+                Files.write(configFile.toPath(), replaced);
+            }
+        }
+    }
+
+    TestXMLConfigurationSource configSource = new TestXMLConfigurationSource();
+
+    private TileLayerConfiguration getConfig(MockWepAppContextRule extensions) throws Exception {
+        extensions.addBean(
+                "MBTilesLayerConfigProvider",
+                new MBTilesLayerXMLConfigurationProvider(),
+                XMLConfigurationProvider.class);
+        return configSource.create(extensions, configDir);
+    }
+
+    @Test
+    public void testLayerDefaultValues() {
```

Comment:
I would add a test that ensures the right tile is picked when fetching a given x/y/z, to avoid y axis orientation issues in the future. Having the mbtiles dependency means you should have the ability to decode a vector tile and look at its contents.